packages/flutter/lib/src/widgets/router.dart

```diff
@@ -1466,12 +1467,18 @@ class PlatformRouteInformationProvider extends RouteInformationProvider with Wid
     required RouteInformation initialRouteInformation,
   }) : _value = initialRouteInformation;
 
+  static bool _equals(Uri a, Uri b) {
+    return a.path == b.path
```

Comment:
What about all the other properties a URI can have? E.g. host, scheme, etc? If the only problem is that query params are out of order, would it make sense to remove the query params from both URIs, compare that those URIs are equal and then check the querParams separately?

Reply:
host and scheme only make sense if this is coming from the engine. I can't find a reasonable use case that the state in framework can translate into host and scheme. That means the host and scheme is always empty when framework reports uri to engine.

If we compares the host and scheme here, we would send the uri back with `replace: false` if the path/queryparam/fragment is the same but host and scheme are different. I think that would confuse the engine.

Reply:
we could always check for the host and scheme in the future if we identify a use case

Reply:
If we compare host and scheme now, this would only affect mobile platform as the host and scheme will only be filled out if this is a deeplink.

So we would immediately send a `replace: false` with the same uri but empty host and scheme to engine. Which currently is a noop for non web platform. so either way wouldn't change the behavior, but if people listen to the navigation channel for some reason, they may get some confusing result.